rct229/rulesets/ashrae9012019/ruleset_functions/get_aggregated_zone_hvac_fan_operating_schedule.py

```diff
@@ -0,0 +1,44 @@
+from pydash import map_
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_list_hvac_systems_associated_with_zone import (
+    get_list_hvac_systems_associated_with_zone,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.utility_functions import find_exactly_one_schedule
+
+
+def get_aggregated_zone_hvac_fan_operating_schedule(rmi, zone_id):
+    """
+     This function loops through all of the HVAC system fan operating schedules associated with a specific zone and
+     creates an aggregated fan operating schedule for the zone. More specifically, if any of the fan operating schedules
+     associated with any of the hvac systems serving the zone have a 1 for a particular hour of the year then the aggregated schedule will equal 1
+     for that particular hour of the year. The function will check this for each hour of the year and return an 8760 aggregated fan operating schedule.
+
+    Parameters
+    ----------
+    rmi: dict A zone id associated with the RMR for which to create the aggregated fan operating schedule as described above.
+    zone_id: str A zone id associated with the RMR for which to create the aggregated fan operating schedule as described above.
+
+    Returns
+    ----------
+    aggregated_zone_hvac_fan_operating_schedule_x: 8760 aggregated fan operating schedule for the zone.
+    """
+
+    aggregated_zone_hvac_fan_operating_schedule = {}
+    for HVAC_id in get_list_hvac_systems_associated_with_zone(rmi, zone_id):
+        for fan_sys in find_all(
+            f'$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*][?(@.id = "{HVAC_id}")].fan_system',
+            rmi,
+        ):
+            fan_sch = getattr_(
+                find_exactly_one_schedule(
+                    rmi, getattr_(fan_sys, "fan_system", "operating_schedule")
+                ),
+                "schedules",
+                "hourly_values",
+            )
+            aggregated_zone_hvac_fan_operating_schedule[fan_sys["id"]] = map_(
+                fan_sch, lambda x: 1.0 if x == 1 else 0
+            )
```

Comment:
1. don't use pydash `map_`. Its performance is not so great compare to the python native `map` function
2. I am not sure this is how the logic is. My understanding is at any hour if one HVAC is off, then the `aggregated_zone_hvac_fan_operating_schedule` at that hour is 0. So a case like this:
```python
schedules = [
     [1] * 8670 # hvac 1
     [1] * 8760 # hvac 2
]
schedule[1][0] = 0
```
will generate an `aggregated_zone_hvac_fan_operating_schedule = [0, 1, 1, 1,....]`

I recommend you saving all hourly value schedules into a 2-dimension list, and then use pandas to do the tranformation:
```python
df = pd.DataFrame(schedules)
df = df.astype(int) # make sure fractions are 0
flattened_list = (df.sum(axis=0) == len(df)).astype(int).tolist()
```

Reply:
This is great! Thank you Weili!